cmd/lakectl/cmd/root.go

```diff
@@ -216,25 +216,49 @@ func withSyncFlags(cmd *cobra.Command) {
 	withNoProgress(cmd)
 }
 
+func getStorageConfig(ctx context.Context, client *apigen.ClientWithResponses, repositoryID string) (*apigen.StorageConfig, error) {
+	confResp, err := client.GetConfigWithResponse(ctx)
+	DieOnErrorOrUnexpectedStatusCode(confResp, err, http.StatusOK)
+
+	storageConfigList := confResp.JSON200.StorageConfigList
+	if len(*storageConfigList) > 1 {
+		repoResp, errRepo := client.GetRepositoryWithResponse(ctx, repositoryID)
+		DieOnErrorOrUnexpectedStatusCode(repoResp, errRepo, http.StatusOK)
+		storageID := repoResp.JSON200.StorageId
+		// find the storage config for the repository
+		for _, storageConfig := range *storageConfigList {
+			if storageConfig.BlockstoreId == storageID {
+				return &storageConfig, err
+			}
+		}
+		Die("Storage config not found for repo "+repositoryID, 1)
+		return nil, nil // this is unreachable - just to make the go code happy
+	} else {
+		storageConfig := confResp.JSON200.StorageConfig
```

Comment:
nit: this function can be a bit cleaner returning immediately if there is no list and continuing otherwise without the else